### PR TITLE
Add basic CIRCT compiler support.

### PR DESCRIPTION
**Pull request by mikeurbach**
This adds basic support for CIRCT as requested in #3825. The addition
is based on the similar additions for MLIR in #3733 and #3770.

### CODE REVIEW

**Comment by mikeurbach** [comment]
Not sure who to request review from, maybe @partouf?

File: etc/config/circt.amazon.properties [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+supportsBinary=false
+supportsExecute=false
+supportsAsmDocs=false
+
+group.circtopt.compilers=circtopt-trunk-20220713
+group.circtopt.isSemVer=false
+group.circtopt.baseName=CIRCT opt
+
+compiler.circtopt-trunk-20220713.exe=/opt/compiler-explorer/circt-circt-trunk-20220713/bin/circt-opt
```
Comment (partouf) [inline etc/config/circt.amazon.properties:13]: I suspect this is just going to be /opt/compiler-explorer/circt-trunk/ and it will be refreshed every day

Comment (partouf) [reply etc/config/circt.amazon.properties:13]: so it could also just be called circtopt-trunk

**Comment by partouf** [comment]
So you mentioned here https://github.com/compiler-explorer/compiler-explorer/issues/3825#issuecomment-1175418933 that it should be fed through mlir-opt first, and then through circt-opt. Is this equivalent to that process?

**Comment by mikeurbach** [comment]
> So you mentioned here https://github.com/compiler-explorer/compiler-explorer/issues/3825#issuecomment-1175418933 that it should be fed through mlir-opt first, and then through circt-opt. Is this equivalent to that process?

This is just to enable feeding the IRs CIRCT knows about through circt-opt on its own. Already this enables some interesting use-cases that the CIRCT community might be interested in (there are many passes in circt-opt already, and "exploring" those seems generally useful).

I haven't started looking into what it would look like to feed some input to mlir-opt first, and then feed the output IR through circt-opt. If that requires further additions, I guess that could be a follow up addition?

File: etc/config/circt.amazon.properties (hunk 1 above)
Comment (mikeurbach) [reply etc/config/circt.amazon.properties]: I pushed up a commit to rename that and change the path to `/opt/compiler-explorer/circt-trunk/`. I wasn't sure exactly what the path should be. Is that something I can find in compiler-explorer/infra CI logs or something?

**Comment by partouf** [comment]
This is now live! Please have a look around and let us know your thoughts

**Comment by mikeurbach** [comment]
Looks great! Thanks! I will go ahead and close the initial issue I opened, and open smaller discussions for further enhancements.

**Pull request merged**